### ISSUE TITLE
Using with goprotobuf

### ISSUE THREAD

**Issue by Kalthorn** [kuroneko/gosqlite3#8]
Hi, thanks for all the work you've put into this package so far!

My question is: how do I retrieve protobufs that have been stored in a sqlite database as (medium) blobs?  It probably isn't relevant to the answer, but I didn't store these protobufs myself, otherwise I would probably know a lot more about the process.

I've read helpers_test.go and that taught me how to retrieve general values from the database and I see that blobs have a type of *gob.Decoder.  The 3rd column in my rows are the blobs and helpers_test.go uses a type selector on the returned interface values like so:

```
data := values[2]
switch data := data.(type) { ... }
```

Are all blobs in gosqlite3 treated as gobs?  Does it matter (ie, can any complex type be treated as a gob?) Should I not switch on the type and retrieve the value from the database in some other way?  Looking at the code for ResultColumn.Value() leads me to believe that all blobs are returned as gob decoders.

Basically I don't know if I need to Decode() the value from a gob and then Unmarshal the protobuf from that or if I should Unmarshal right away.  The original database was populated from a different language so gobs were never used.  I'm also sure the original protobufs were from version 2.01 so my attempts here may be incompatible for other reasons.

**Comment by feyeleanor** [comment]
The current API assumes that blobs are only used to store binary objects created by a go program and serialised using a gob encoder. This was a short-sighted assumption on my part as I wanted a quick and easy way to store binary objects in the database without writing a full-blown ORM.

In your case the data stored in the database isn't gob encoded so the current API isn't adequate to handle that and needs to be changed. I'll need a couple of days to think over the best way to handle this and write an appropriate patch.

I can't unfortunately help you with the protobufs side of your problem as I've not played with them. However if you come and ask on the go-nuts mailing list I'm sure someone there will be able to steer you in the right direction.

**Comment by Kalthorn** [comment]
Ahh alright.  No need to change the API on my account though, unless this is something you or others have wanted anyway.  I'll take a look at Russ's super lightweight sqlite wrapper since my project really is a short, one-off experiment.  Thanks for the quick response!

**Comment by feyeleanor** [comment]
It's okay, I plan to make the API a bit more ORM-friendly anyway so looking at alternatives whilst we still have a small user-base is probably a good idea.
